File: lab1/entdoctor.cpp
#include "entdoctor.h"
#include <iostream>
#include <string>
#include <vector>
#include "datetime.h"
using namespace std;

ENTDoctor::ENTDoctor(const string& name, const string& specialization) : Doctor(name, specialization) {}

void ENTDoctor::addSchedule(const pair<int, int> time) {
    schedule.push_back(time);
}

void ENTDoctor::displayInfo() const {
    cout << "Name: " << getName() << endl;
    cout << "Specialization: " << specialization << endl;
    cout << "Schedule:" << endl;
    for (const auto& scheduleItem : schedule) {
        cout << getDayOfWeek(scheduleItem.first) << " at " << scheduleItem.second << ":00" << endl;
    }
}

void ENTDoctor::bookAppointment(const DateTime& time) {
    cout << "Appointment booked with " << getName() << " at " << formatDate(time) << endl;
}

const vector<pair<int, int>>& ENTDoctor::getSchedule() const {
    return schedule;
}

File: Models/ENTdoctor.cs
﻿namespace Project.Models;

using System;
using System.Collections.Generic;

public class EnTdoctor(string name, string specialization) : Doctor(name, specialization)
{
    private List<KeyValuePair<int, int>> _schedule = new();

    public void AddSchedule(KeyValuePair<int, int> time)
    {
        _schedule.Add(time);
    }

    public override void DisplayInfo()
    {
        Console.WriteLine("Name: " + GetName());
        Console.WriteLine("Specialization: " + specialization);
        Console.WriteLine("Schedule:");
        foreach (var scheduleItem in _schedule)
        {
            Console.WriteLine($"{GetDayOfWeek(scheduleItem.Key)} at {scheduleItem.Value}:00");
        }
    }

    public override void BookAppointment(Data time)
    {
        Console.WriteLine($"Appointment booked with {name} at {time}");
    }

    public override List<KeyValuePair<int, int>> GetSchedule()
    {
        return _schedule;
    }
    
}
